docs/rules/no-boolean-conditional.md

```diff
@@ -0,0 +1,28 @@
+# Disallow boolean literals in conditional expressions (no-boolean-conditional)
+
+Its a common mistake in javascript to evaluate an conditional expression and then return `Boolean` back and not realizing that the evaluation of conditional expression implicitly returns a 'Boolean'.
```

Comment:
> It's a common mistake in JavaScript to use a conditional expression to select between two Boolean values instead of using `!` to convert the test to a Boolean.